feature: allow sync flag to be passed to ThreadStream for synchronous…

karankraina opened the pull request:
Allow `sync` flag to be passed to ThreadStream for sync logging.
Ref - #1801

karankraina commented:
@mcollina Do you think this works ?

mcollina reviewed (changes requested):
Thanks for opening a PR! Can you please add a unit test?

karankraina commented:
@mcollina Added unit test.

mcollina reviewed (approved):
lgtm

mcollina commented on test/transport/syncTrue.test.js:
```diff
@@ -0,0 +1,55 @@
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
+  instance.info(value)
```
I think you could do `instance.flushSync()` and then just try reading the file. It should work.

karankraina replied:
Wouldn't calling `flushSync` force synchronously flush all messages to the stream ? That would pass even if `sync` is `false` ? I tried to simulate a event loop block immediately after calling `instance.info`. I think I need to update the loop duration by some bit. Am i missing something?

mcollina replied:
ah no prob. This is ok

karankraina replied:
Updated PR. Should pass now :)

karankraina replied:
@mcollina can you retrigger the CI actions please

mcollina commented:
You might need to pass `flush: true` to https://nodejs.org/api/fs.html#filehandlecreatewritestreamoptions in https://github.com/pinojs/pino/blob/f220965c3e3885b8479c37ec880e43315e1a2373/test/fixtures/to-file-transport.js#L8. Otherwise Node.js would not `fsync` the file descriptor, so you might not be able to read it.

karankraina commented:
@mcollina Ah yess, updated PR. `flush: true` fixed it.  Added loop duration by a bit just in case.

karankraina commented:
> @mcollina Ah yess, updated PR. `flush: true` fixed it. Added loop duration by a bit just in case.

Ubuntu 22 failing for some reason. :( Need to re-check this. Had worked on my fork
